server/controllers/finance/reports/cashReport/index.js

```diff
@@ -0,0 +1,188 @@
+const ReportManager = require('../../../../lib/ReportManager');
+const BadRequest = require('../../../../lib/errors/BadRequest');
+
+const TEMPLATE = './server/controllers/finance/reports/cashReport/report.handlebars';
+
+// expose to the API
+exports.document = document;
+
+
+function getRecordQuery (token){
+  const query =
+  `SELECT
+    t.trans_id, t.trans_date, t.debit_equiv AS debit, t.credit_equiv AS credit, t.description, 
+		t.origin_id, t.user_id, u.username, a.number, 
+		tr.text AS transactionType, a.label
+  FROM
+    (
+      (
+      SELECT
+        p.trans_date, p.debit_equiv, p.credit_equiv,
+        p.trans_id, p.description, p.origin_id, p.user_id, p.account_id
+      FROM
+        posting_journal AS p
+      WHERE 
+			p.account_id= ? AND (p.trans_date >= DATE(?) AND p.trans_date <= DATE(?))
+      )
+      UNION ALL
+      (
+      SELECT
+        g.trans_date, g.debit_equiv, g.credit_equiv,
+        g.trans_id, g.description, g.origin_id, g.user_id, g.account_id
+      FROM
+        general_ledger AS g
+      WHERE 
+			g.account_id= ? AND (g.trans_date >= DATE(?) AND g.trans_date <= DATE(?))
+      )
+    ) AS t
+    JOIN 
+	 	user u ON t.user_id = u.id
+    JOIN 
+	 	account a ON a.id = t.account_id
+    LEFT JOIN 
+	 	transaction_type tr ON tr.id = t.origin_id
+    WHERE ${token} GROUP BY t.trans_id;`
+
+  return query;
+}
+
+function aggregateRecordQuery (token = 1){
+  const query =
+  `
+  SELECT
+    ABS(SUM(t.debit - t.credit)) AS balance
+  FROM
+    (
+      (
+      SELECT
+        p.debit_equiv AS debit, p.credit_equiv AS credit
+      FROM
+        posting_journal AS p
+      WHERE 
+        p.account_id= ? AND 
+        (p.trans_date >= DATE(?) AND p.trans_date <= DATE(?))
+      )
+      UNION ALL
+      (
+      SELECT
+        g.debit_equiv AS debit, g.credit_equiv AS credit
+      FROM
+        general_ledger AS g
+      WHERE 
+			  g.account_id= ? AND 
+        (g.trans_date >= DATE(?) AND g.trans_date <= DATE(?))
+      )
+    ) AS t WHERE ${token};`
+
+  return query;
+}
+
+function getCashRecord (accountId, dateFrom, dateTo){
+  let reportContext = {};
+
+  // Getting enries records
+  return db.exec(getRecordQuery('t.debit_equiv > 0'), [accountId, dateFrom, dateTo, accountId, dateFrom, dateTo])
+    .then((entries) => {
+      reportContext.entries = entries;
+
+      // Getting expenses records
+      return db.exec(getRecordQuery('t.credit_equiv > 0'), [accountId, dateFrom, dateTo, accountId, dateFrom, dateTo]);
+    })
+    .then((expenses) => {
+      reportContext.expenses = expenses;
+
+      // Getting sum entries 
+      return db.one(aggregateRecordQuery('t.debit > 0'), [accountId, dateFrom, dateTo, accountId, dateFrom, dateTo]);      
+    })
+    .then((totalEntry) => {
+      reportContext.totalEntry = totalEntry.balance;
+
+      // Getting sum expenses 
+      return db.one(aggregateRecordQuery('t.credit > 0'), [accountId, dateFrom, dateTo, accountId, dateFrom, dateTo]);   
```

Comment:
There ~5 calls to posting_journal UNION general_ledger.  Since you always look at `account_id` , you could replace some of the sums on the `general_ledger` with `period_total`s?